plugins/outputs/splunk/splunk.go

```diff
@@ -0,0 +1,277 @@
+			req, err := http.NewRequest("POST", s.SplunkUrl, bytes.NewBuffer(payload))
+			if err != nil {
+				return fmt.Errorf("unable to create http.Request \n    URL:%s\n\n", s.SplunkUrl)
+			}
+
+			req.Header.Add("Content-Type", "application/json")
+			req.Header.Add("Authorization", "Splunk "+s.AuthString)
+
+			//// Check for existence of s && req to prevent race ('panic: runtime error: invalid memory address or nil pointer dereference')
+			if s != nil && req != nil {
+				resp, err := s.client.Do(req)
+				if err != nil {
+					return fmt.Errorf("error POST-ing metrics to Splunk[%s]  Sending Data:%s\n", err, payload)
+				}
+
+				defer resp.Body.Close()
+
+				if resp.StatusCode < 200 || resp.StatusCode > 209 {
+					return fmt.Errorf("received bad status code posting to %s, %d\n\n%s\n", s.SplunkUrl, resp.StatusCode, payload)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Check for metric name ending with .value and remove it, sanitize characters and add separators
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldName(measureName string, fieldName string, s *Splunk) string {
+	var name string
+	if !s.SimpleFields && fieldName == "value" {
+		name = fmt.Sprintf("%s%s", s.Prefix, measureName)
+	} else {
+		name = fmt.Sprintf("%s%s%s%s", s.Prefix, measureName, s.MetricSeparator, fieldName)
+	}
+
+	if s.UseRegex {
+		name = sanitizedRegex.ReplaceAllLiteralString(name, "-")
+	} else {
+		name = sanitizedChars.Replace(name)
+	}
+
+	if s.ConvertPaths {
+		name = pathReplacer.Replace(name)
+	}
+	return name
+}
+
+// -----------------------------------------------------------------------------------------------------------------------
+//  Parse the value depending on the data type
+// -----------------------------------------------------------------------------------------------------------------------
+func processFieldValue(v interface{}, name string, s *Splunk) (float64, error) {
+	switch p := v.(type) {
+	case bool:
+		if s.ConvertBool {
+			if p {
+				return 1, nil
+			} else {
+				return 0, nil
+			}
+		}
+	case int64:
+		return float64(v.(int64)), nil
+	case uint64:
+		return float64(v.(uint64)), nil
+	case float64:
+		return v.(float64), nil
+	case string:
+		for prefix, mappings := range s.StringToNumber {
+			if strings.HasPrefix(name, prefix) {
+				for _, mapping := range mappings {
+					val, hasVal := mapping[string(p)]
+					if hasVal {
+						return val, nil
+					}
+				}
+			}
+		}
+		return 0, fmt.Errorf("unexpected type: %T, with value: %v, for: %s", v, v, name)
+	default:
+		return 0, fmt.Errorf("unexpected type: %T, with value: %v, for: %s", v, v, name)
+	}
+
+	return 0, fmt.Errorf("unexpected type: %T, with value: %v, for: %s", v, v, name)
+}
```

Comment:
There's lots of other numeric types being missed: `int`, `uint`, `uint8`, `uint16`, `uint32`, `int8`, `int16`, `int32`, `float32`.

Reply:
added the additional types.